lib/dat-worker-pool.rb

```diff
@@ -1,67 +1,47 @@
 require 'logger'
-require 'system_timer'
-require 'thread'
 
 require 'dat-worker-pool/version'
 require 'dat-worker-pool/default_worker'
 require 'dat-worker-pool/queue'
+require 'dat-worker-pool/runner'
 
 class DatWorkerPool
 
   DEFAULT_NUM_WORKERS = 1
   MIN_WORKERS         = 1
 
-  attr_reader :num_workers, :logger, :queue
-  attr_reader :on_worker_error_callbacks
-  attr_reader :on_worker_start_callbacks, :on_worker_shutdown_callbacks
-  attr_reader :on_worker_sleep_callbacks, :on_worker_wakeup_callbacks
-  attr_reader :before_work_callbacks, :after_work_callbacks
+  attr_reader :logger, :queue
 
   def initialize(options = nil)
     options ||= {}
-    @do_work_proc = options[:do_work_proc]
-    @num_workers  = (options[:num_workers] || DEFAULT_NUM_WORKERS).to_i
-    @logger       = options[:logger] || NullLogger.new
-
-    if @num_workers < MIN_WORKERS
+    num_workers = (options[:num_workers] || DEFAULT_NUM_WORKERS).to_i
+    if num_workers < MIN_WORKERS
       raise ArgumentError, "number of workers must be at least #{MIN_WORKERS}"
     end
 
+    @logger = options[:logger] || NullLogger.new
+
     @queue = options[:queue] || begin
       require 'dat-worker-pool/default_queue'
       DatWorkerPool::DefaultQueue.new
     end
 
-    @workers_waiting = WorkersWaiting.new
-    @mutex           = Mutex.new
-    @workers         = []
-
-    @on_worker_error_callbacks    = []
-    @on_worker_start_callbacks    = []
-    @on_worker_shutdown_callbacks = [proc{ |worker| despawn_worker(worker) }]
-    @on_worker_sleep_callbacks    = [proc{ @workers_waiting.increment }]
-    @on_worker_wakeup_callbacks   = [proc{ @workers_waiting.decrement }]
-    @before_work_callbacks        = []
-    @after_work_callbacks         = []
-
-    @started = false
+    # TODO - don't save off once a worker class can be passed in
+    @worker_class = DefaultWorker
```

Comment:
@kellyredding - We have to save off the worker for now because of the callback methods, this will eventually be removed once we start passing in a worker class.